### DISCUSSION TITLE
Guide on using the socket

### DISCUSSION THREAD

**Discussion by karevan** [swoole/swoole-src#4617] [Q&A] [answered]
Hi
I have a question about using the socket
I know that sockets can be used in two ways:
1-use with [Coroutine Socket Client](https://openswoole.com/docs/modules/swoole-coroutine-socket-client)
2-use with [WebSocket Server](https://openswoole.com/docs/modules/swoole-websocket-server)
But I do not know the **differences** between these two methods!
Does anyone guide me?
Thanks to everyone

**Comment by leocavalcante** [comment, marked as answer]
There are two main differences here
1. Client vs Server
1. Socket vs WebSocket

## Clients vs Server
The difference here is almost the same difference between a webserver and a web-browser. The server will be up, bound to a host and port, where the client can connect to.

## Socket vs WebSocket
Raw sockets over TCP/UPD/Unix are very low-level and doesn't provides some protocol guarantees like WebSocket.
WebSocket is the one where you can access from a web-browser.

---

Another tip: refer to [wiki.swoole.com](https://wiki.swoole.com/#/) for the official docs (right-click then Translate to English).
1. Raw socket:
    1. Coroutine server: https://wiki.swoole.com/#/server/co_init
    1. Coroutine client: https://wiki.swoole.com/#/coroutine_client/init
1. WebSocket:
    1. Coroutine style
        1. Server: https://wiki.swoole.com/#/coroutine/ws_server
        1. Client: https://wiki.swoole.com/#/coroutine_client/http_client
    1. **Async style (the one you probably wants to be using)**
        1. Server: https://wiki.swoole.com/#/websocket_server

**Reply by karevan** [reply]
Thank you so much for your help
